### PR TITLE
Add teamcity support v2 + refactoring

### PR DESCRIPTION
**Pull request by agentram**
Hi,

I made some updates for TeamCity output and tried to refactor config generation. Sorry if this is too much of a change - when I started, one thing pulled another and so on...

This is how TC build looks now:
Screenshot 2021-01-28 at 11 49 00
It contains information about all tested libraries (total number of tests) and failed tests for libraries with vulnerabilities. You can also see a description of the vulnerability in the test output.

### CODE REVIEW

File: cmd/dtrack-audit/main.go [hunk 1]
```diff
@@ -1,134 +1,74 @@
 package main
 
 import (
-	"encoding/json"
-	"flag"
+	"errors"
 	"fmt"
-	"github.com/ozonru/dtrack-audit/internal/dtrack"
+	"github.com/agentram/dtrack-audit/internal/dtrack"
```
Comment (agentram) [inline cmd/dtrack-audit/main.go:6]: Please note that I changed the import path to have the ability to run go get from my branch:
```
go get github.com/agentram/dtrack-audit/cmd/dtrack-audit@add_teamcity_support_v2
```
We can put it back when it's ready to merge.

File: cmd/dtrack-audit/teamcity.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"github.com/agentram/dtrack-audit/internal/dtrack"
```
Comment (agentram) [inline cmd/dtrack-audit/teamcity.go:7]: Please note that I changed the import path to have the ability to run go get from my branch:
```
go get github.com/agentram/dtrack-audit/cmd/dtrack-audit@add_teamcity_support_v2
```
We can put it back when it's ready to merge.

**Comment by oxdef** [comment]
Hi!

Thanks for the MR and sorry for delay. If you don't mind I suggest to split into small ones: config and TeamCity support v2.

**Comment by agentram** [comment]
No problems, I will prepare a MR for config soon.

**Comment by agentram** [comment]
I will merge https://github.com/ozonru/dtrack-audit/pull/11 when it's approved

**Comment by oxdef** [comment]
@agentram  Sorry for the delay. I've just merged config PR and made some changes in release branch. Could you please move teamcity support code  into `/internal/dtrack/` package like for https://github.com/ozonru/dtrack-audit/blob/release/v1.0.0/internal/dtrack/config.go?

**Comment by agentram** [comment]
Hi @oxdef 
I have updated the PR. Let me test it in my builds for a few days.

**Comment by agentram** [comment]
@oxdef Everything is alright. You can review it.

**Review by oxdef — CHANGES_REQUESTED** [review]

File: cmd/dtrack-audit/main.go [hunk 3]
```diff
@@ -64,37 +37,11 @@ func main() {
 		os.Exit(1)
 	}
 
-	uploadResult, err := apiClient.Upload(config.InputFileName, config.ProjectId)
```
[the remark this answers is not in the document]
Comment (agentram) [reply cmd/dtrack-audit/main.go:67]: Roger! I moved it back.

File: cmd/dtrack-audit/main.go [hunk 4]
```diff
@@ -64,37 +37,11 @@ func main() {
...
-			}
-			if config.UseTeamCityOutput {
-				printTeamCityMsg("fail", "")
-			}
-			os.Exit(1)
-		}
+	if config.SyncMode {
 		if config.UseTeamCityOutput {
-			printTeamCityMsg("pass", "")
+			dtrack.PrintForTeamCity(apiClient, config)
```
[the remark this answers is not in the document]
Comment (agentram) [reply cmd/dtrack-audit/main.go]: This is done!

File: cmd/dtrack-audit/main.go [hunk 5]
```diff
@@ -64,37 +55,13 @@ func main() {
...
-		fmt.Printf("SBOM file is successfully uploaded to DTrack API. Result token is %s\n", uploadResult.Token)
-	}
-
-	if uploadResult.Token != "" && config.SyncMode {
-		if config.UseTeamCityOutput {
-			printTeamCityMsg("run", "")
-		}
-		err := apiClient.PollTokenBeingProcessed(uploadResult.Token, time.After(time.Duration(config.Timeout)*time.Second))
+	if config.SyncMode {
+		findings, err := findVulnerabilities(apiClient, config)
```
Comment (oxdef) [inline cmd/dtrack-audit/main.go:59]: You broken here Async mode. I will fix it in release branch.

**Review by oxdef — APPROVED** [review]
@agentram 
Thanks for improved version of TeamCity support!

**Pull request merged**